### ISSUE TITLE
Any chance of having a mod for the Plug and Play 2 Controller?

### ISSUE THREAD

**Issue by SixelAlexiS90** [MarcRiera/ddgo-pnp-controller#4]
Hi! First of all thank you for this amazing mod for the first Plug and Play :)

I ask this since it would be really great to have a similar mod the Plug and Play 2 Controller (Shinkansen Ex).

Various months ago I've asked about this to GMMan and this was their response:

_`I would like to make a mod for PnP2, but it's pretty low on the priority list, and a major challenge is the installation process. While the original could live modify the system, the new one doesn't have such a system (instead opting for signed update packages), so it requires booting from FEL and coordinating that is a bit of a tricky thing that no one's really satisfactorily solved yet.`_

Have you tried to do anything with it or you are of the same advice?

Thank you again :)

**Comment by marcriera** [comment]
Hello! Thanks for the support!

I would also like to have a similar mod for the PnP2, but unfortunately I don't own one and I'm not interested in it besides the mod at the moment. This might change in the future, but it's the honest situation currently.

Regarding the technical details:

1. I followed GMMan's analysis of the device and I agree it's a bit tricky. Because we can't sign update packages, we need FEL mode, which is a mode where the device connects to a PC via USB and exposes the internal storage directly. So we would need to run some sort of installer on the PC. I have a little experience with FEL on the original PnP (which also has it) and I've never managed to get everything right and fully working.

2. Once a reliable way to modify the system is found, the custom software is easy. The mod for the original PnP was a bit tricky to set up because the device uses an old Linux kernel. The "adapter" software works both on old and recent versions of the Linux kernel and can emulate USB devices on both. Only minor changes are required: reading input from the physical handles and buttons and emulating other controllers in a certain way. The current software is modular and flexible enough to support this easily.

To sum up: we need to do two things. The first part is tricky and I don't have all the details (or the device) needed to do it, and the second part is easy. So yes, I agree with GMMan.

Thanks!
